From: Ben Hutchings <ben@decadent.org.uk>
Subject: perf: Hide ABI change in 3.2.30
Forwarded: not-needed

<linux/perf_events.h> is included by many source files but we don't
want to bump the ABI number again right now.

The new member (atomic_long_t refcount) has the same size as the old
(struct file *filp) and neither should be needed by OOT modules (if
they make use of perf_events at all).  So pretend this didn't happen.
Index: linux-3.2.46/include/linux/perf_event.h
===================================================================
--- linux-3.2.46.orig/include/linux/perf_event.h	2013-07-26 18:54:05.000000000 +0000
+++ linux-3.2.46/include/linux/perf_event.h	2013-07-26 19:24:59.000000000 +0000
@@ -794,7 +794,11 @@
 	struct hw_perf_event		hw;
 
 	struct perf_event_context	*ctx;
+#ifdef __GENKSYMS__
+	struct file			*filp;
+#else
 	atomic_long_t			refcount;
+#endif
 
 	/*
 	 * These accumulate total time (in nanoseconds) that children
